northstar/src/runtime/process/trampoline.rs

```diff
@@ -0,0 +1,26 @@
+use super::{init::Init, Checkpoint};
+use crate::runtime::pipe::Channel;
+use nix::{sched, unistd};
+use std::process::exit;
+
+pub(super) fn trampoline(init: Init, mut child_channel: Channel, checkpoint_init: Checkpoint) -> ! {
```

Comment:
I have the impression that this pattern is repeated in a couple of places. Would it be possible to have a generalization of this like:

```rust
fn spawn_child(task: impl FnOnce() -> Result<(), Error>) -> impl Future<Output = Result<ExitStatus, Error>> {
    match unsafe { fork() }.expect("...") {
        Parent { child } => {
            waitpid(child)
        }
        Child => {
            match task() {
                Ok(()) => exit(0),
                Err(e) => // log or something and exit with -1 or something
            }
        }
    }
}
```

Reply:
I think the three forks are very different and probably easier to read with a match for each since we read this `if fork() == -1 {.....` since years :-)